src/V3Gate.cpp

```diff
@@ -476,6 +476,11 @@ class GateVisitor final : public GateBaseVisitor {
     virtual void visit(AstCFunc* nodep) override {
         iterateNewStmt(nodep, "User C Function", "User C Function");
     }
+    virtual void visit(AstCAwait* nodep) override {
+        iterateChildren(nodep);
+        VL_RESTORER(m_logicVertexp);
+        if (nodep->sensesp()) iterateNewStmt(nodep->sensesp(), "Timing domain", "Timing domain");
+    }
```

Comment:
Why do we need to iterate the sentreep() as as separate block? In here - relatively old code - "NewStmt" means roughly a logic block, just like in V3Order, e.g.: a continuous assignment of an always block. The AstCAwait is already under such logic block, so it is a bit suspect, can you elucidate please?

Reply:
This was a quick fix, not sure yet how to do a proper one. I put this here because for a case like this:
```systemverilog
assign inv_clk = ~clk;
initial begin
    ...
    @(inv_clk);
```
`V3Gate` removes the `assign`.

Reply:
I guess this is because you made the sentrees unique in V3Timing, which is before V3Gate, and not in V3ActiveTop where the rest are made unique, and is after V3Gate. I think you have 2 options:

1. Put the sen-trees as children in AstCAwait, then it should all just work, but need to make them unique in V3ActiveTop.
2. Iterate the global sen-trees (the ones directly under AstTopScope) themselves in V3Gate as separate logic block (i.e.: invoke `iterateNewStmt`).

Option 2 is closer to what you have now, I'm happy with either, so your choice.

Reply:
Ended up moving `V3Timing` after `V3Gate`, as the latter can modify sentrees. And I need to uniquify them in `V3Timing` (or at least be able to group together equivalent ones).